### PR TITLE
Adding scraping to OSM pod

### PR DESCRIPTION
**Pull request by eduser25**
OSM has since recently started exporting prometheus format metrics.
This commit adds scrapping annotation on the OSM deployment, which will
be automatically picked up by Prometheus's Pod-based scrapping.

Also added Injector histogram and Connected proxies on the Mesh details
dashboard.

**Affected area**:

- Metrics                [X]

- Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?
No

### CODE REVIEW

File: charts/osm/grafana/dashboards/osm-mesh-envoy-details.json [hunk 1]
```diff
@@ -435,13 +434,19 @@
...
+          "legendFormat": "OSM Proxy Connect",
+          "refId": "B"
         }
       ],
       "thresholds": [],
       "timeFrom": null,
       "timeRegions": [],
       "timeShift": null,
-      "title": "Total number of pods in Mesh",
+      "title": "Pods in Mesh vs Connected Proxies",
```
Comment (shashankram) [inline charts/osm/grafana/dashboards/osm-mesh-envoy-details.json:449]: This title might not make much sense to users, is there something else we can put in here?

File: charts/osm/grafana/dashboards/osm-mesh-envoy-details.json [hunk 2]
```diff
@@ -1004,23 +879,17 @@
...
-          "legendFormat": "",
-          "refId": "A"
         }
       ],
       "thresholds": [],
       "timeFrom": null,
       "timeRegions": [],
       "timeShift": null,
-      "title": "Rejected Webhooks",
+      "title": "OSM injector webhooks",
```
Comment (shashankram) [inline charts/osm/grafana/dashboards/osm-mesh-envoy-details.json:892]:
```suggestion
      "title": "OSM sidecar injector",
```

**Review by shashankram — APPROVED** [review]

**Review by snehachhabria — APPROVED** [review]

File: charts/osm/grafana/dashboards/osm-mesh-envoy-details.json (hunk 1 above)
Comment (eduser25) [reply charts/osm/grafana/dashboards/osm-mesh-envoy-details.json:449]: Let me revert this one. I struggled to give it a name; I think you are right.

Comment (eduser25) [inline charts/osm/grafana/dashboards/osm-mesh-envoy-details.json:449]:
```suggestion
      "title": "Total number of pods in Mesh",
```

**Pull request merged**